src/Connector.SqlServer/Utils/Upgrade/UpgradeCodeTableUtility.cs

```diff
@@ -0,0 +1,74 @@
+﻿using CluedIn.Connector.SqlServer.Connector;
+using CluedIn.Connector.SqlServer.Utils.TableDefinitions;
+using CluedIn.Core.Streams.Models;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CluedIn.Connector.SqlServer.Utils.Upgrade
+{
+    internal class UpgradeCodeTableUtility
+    {
+        public static async Task AddIsDataPartOriginEntityCodeToTable(IReadOnlyStreamModel streamModel, SqlName schema, SqlTransaction transaction, ILogger logger)
+        {
+            var codeTableName = TableNameUtility.GetCodeTableName(streamModel, schema);
+            var codeTableColumnsSqlQueryText = $"""
+                SELECT [COLUMN_NAME]
+                FROM INFORMATION_SCHEMA.COLUMNS
+                WHERE TABLE_NAME = N'{codeTableName.LocalName}'
```

Comment:
Review whether schema should also be compared, as you could have same name in different schemas